### PR TITLE
Adding redux/tooklit and files

### PR DESCRIPTION
**Pull request by Danfall369**
Hi there Code Reviewer

I have completed the following requirements for the project:

Added Redux Toolkit.
- Created a directory that contains all Redux logic.0
- Configured a Redux store.
- Defined a slice of state for books that includes an array of books (initial state: empty array), a reducer that adds a book, and a reducer that removes a book.
- Defined a slice of state for categories that includes an array of categories (initial state: empty array) and a reducer that checks the status and always returns "Under construction" (the initial state should check to that string).
- Structured the application files using a "feature folder" approach and used the ducks pattern for Redux files.
Please review the code changes and let me know if any further changes are needed.

Thank you!

### CODE REVIEW

File: src/redux/categories/categoriesSlice.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+export const categoriesSlice = createSlice({
+  name: 'categories',
+  initialState,
+  reducers: {
+    getStatus: (state) => state.status,
+  },
+});
+
+export default categoriesSlice.reducer;
```
Comment (BenMukebo) [inline src/redux/categories/categoriesSlice.js:16]: - [ ] As you did for the addBook and removeBook, You need to Create 1 reducer, `checkStatus` in the /categories/categoriesSlice.js` ✍🏻 Then Exports by `checkStatus` as an action. Please address this. 👍🏻

**Review by BenMukebo — CHANGES_REQUESTED** [review]
### Hi @Danfall369,

You did a great job so far 👏🏼 👏🏼
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

### To highlight 💪🏻
- Professional README ✔️ 
- Gitflow was correctly used.  ✔️   
- `@reduxjs/toolkit` & `react-redux` are be present in `package.json`✔️ 
- The Redux Store is well configured. ✔️ 
- Actions and reducer for adding and removing books. ✔️ 
 👍  Good job!


## STATUS: CHANGES REQUESTED  ♻️ 
_Please Check the comments under the review and make all required changes_


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- [OPTIONAL] Please it will be nice to add explicit steps to clone and run your project locally, as this will be of help to non-technical users. Please add descriptive steps from cloning your app to seeing the results displayed on the browser. You can put the steps to clone and run the project in a new Getting Started section under the project's description.


Cheers and Happy coding!👏👏👏 💻  
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me @BenMukebo when you comment so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/categories/categoriesSlice.js (hunk 1 above)
Comment (Danfall369) [reply src/redux/categories/categoriesSlice.js:16]: @BenMukebo Done

**Review by didierganthier — APPROVED** [review]
Hi @Danfall369 ,

Congratulations! 🎉 Your project is complete and ready to be merged! 🚀



## Highlights
- Added reducers and actions ✔️ 
- Created reducer `checkStatus` ✔️ 
- Exported `checkStatus` as an action ✔️ 

## Optional Suggestions 💡
- N/A

Cheers, and Happy coding!👏👏👏

If you have any questions or need clarification on anything, feel free to leave a comment in the pull request thread, and don't forget to tag me so I'll receive the notification.

------
As per the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561), you have a limited number of reviews per project. If you feel that the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s). 

Once again, congratulations on completing this project! 🎉

**Pull request merged**